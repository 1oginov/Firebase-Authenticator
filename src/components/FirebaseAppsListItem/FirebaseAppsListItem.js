// @flow

import * as React from 'react';

import type { FirebaseApp } from '../../lib/firebaseApp';

type Props = {
  app: FirebaseApp,
  onSelect: string => void,
};

const FirebaseAppsListItem = ({ app, onSelect }: Props) => (
  <div>
    {app.title}
    <button onClick={() => onSelect(app.id)} type="button">Select</button>
  </div>
);

export default FirebaseAppsListItem;
