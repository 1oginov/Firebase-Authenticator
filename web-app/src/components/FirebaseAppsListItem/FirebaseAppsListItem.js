import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import * as React from 'react';

import type { FirebaseApp } from '../../lib/firebaseApp';

type Props = {
  app: FirebaseApp,
  onSelect: string => void,
};

const FirebaseAppsListItem = ({ app, onSelect }: Props) => (
  <ListItem button onClick={() => onSelect(app.id)}>
    <ListItemText
      primary={app.title ? app.title : app.id}
      primaryTypographyProps={{ noWrap: true }}
    />
  </ListItem>
);

export default FirebaseAppsListItem;
