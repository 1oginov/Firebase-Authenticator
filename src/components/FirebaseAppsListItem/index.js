// @flow

import FirebaseAppsListItem from './FirebaseAppsListItem';

export default FirebaseAppsListItem;
