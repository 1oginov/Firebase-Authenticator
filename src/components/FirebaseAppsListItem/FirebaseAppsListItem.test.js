// @flow

import { shallow } from 'enzyme';
import React from 'react';

import FirebaseAppsListItem from './FirebaseAppsListItem';
import app from '../../lib/__test-data__/firebaseApp';

describe('<FirebaseAppsListItem />', () => {
  it('renders without crashing and matches snapshot', () => {
    const wrapper = shallow(
      <FirebaseAppsListItem
        app={app}
        onSelect={() => undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
