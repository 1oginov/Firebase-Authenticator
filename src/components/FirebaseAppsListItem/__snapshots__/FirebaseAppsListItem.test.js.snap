// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirebaseAppsListItem /> renders without crashing and matches snapshot 1`] = `
<WithStyles(ListItem)
  button={true}
  onClick={[Function]}
>
  <WithStyles(ListItemText)
    primary="App"
    primaryTypographyProps={
      Object {
        "noWrap": true,
      }
    }
  />
</WithStyles(ListItem)>
`;
